recommender/engine/base_agent_recommender.py

```diff
@@ -81,31 +86,61 @@ def _get_user(self, context: Dict[str, Any]) -> User:
 
         return user
 
-    def _for_logged_user(self, user: User, search_data: SearchData, k: int) -> Tuple[int]:
+    def _for_logged_user(
```

Comment:
annotate this class as abstract

Reply:
okey, I will annotate this abstract **method** :)